index.js

```diff
@@ -15,6 +21,71 @@ app.get('/', (req, res) => {
   res.send('Hello World');
 });
 
+// endpoint to get all products
+app.get('/products/all', (req, res) => {
+  var full=[];
+  for (var i in p.products){
+	var obj = {id: p.products[i].id,
+	categoryId: p.products[i].categoryId,
+	name: p.products[i].name,
+	unit: p.products[i].unit,
+	pricePerUnit: p.products[i].pricePerUnit,
+	stocks: p.products[i].stocks,
+	};
+    for (var j in c.categories) {
+        if (p.products[i].categoryId == c.categories[j].id) {
+            obj.categoryName = c.categories[j].categoryName;
+        }
+	}
+	full.push(obj);
+  }
+  res.json(full);
+});
+
+// endpoint to get details of specific product
+app.get('/product/:id', (req, res) => {
+  var obj = {};
+  for (var i in p.products){
+    if(p.products[i].id == req.params.id){
+		obj.id = p.products[i].id;
+		obj.categoryId= p.products[i].categoryId;
+		obj.name= p.products[i].name;
+		obj.unit= p.products[i].unit;
+		obj.pricePerUnit= p.products[i].pricePerUnit;
+		obj.stocks= p.products[i].stocks;
+		for (var j in c.categories) {
+			if (obj.categoryId == c.categories[j].id) {
+				obj.categoryName = c.categories[j].categoryName;
+			}
+		}
+	}
+  }
+  res.json(obj);
+});
+
+// endpoint to get details of specific category
+app.get('/category/:ctyId', (req, res) => {
+  var cat = [];
+  for (var i in p.products){
+    if(p.products[i].categoryId == req.params.ctyId){
+		var obj = {id: p.products[i].id,
+		categoryId: p.products[i].categoryId,
+		name: p.products[i].name,
+		unit: p.products[i].unit,
+		pricePerUnit: p.products[i].pricePerUnit,
+		stocks: p.products[i].stocks,
+		};
+		for (var j in c.categories) {
+			if (obj.categoryId == c.categories[j].id) {
+				obj.categoryName = c.categories[j].categoryName;
+			}
+		}
+		cat.push(obj);
```

Comment:
You can create a single helper function which produces this json obj for you instead of writing the same functionality in all of the methods. That way your code looks cleaner and you reduce the size of your file. Also, don't use variable names like 'i' or 'p' since this doesn't really convey any information to the reader. Always use intuitive variable names which are self descriptive. I hope [this](http://www.informit.com/articles/article.aspx?p=131025&seqNum=3) article is of help to you.